pre_commit/languages/dotnet.py

```diff
@@ -76,9 +76,17 @@ def install_environment(
             ),
         )
 
-        # Cleanup build output
-        for d in ('bin', 'obj', build_dir):
-            rmtree(prefix.path(d))
+        # Remove everything except for the envdir and hook config
+        to_keep = (
+            helpers.environment_dir(ENVIRONMENT_DIR, version),
+            '.pre-commit-hooks.yaml',
+        )
+        for f in os.listdir(prefix.prefix_dir):
+            path = prefix.path(f)
+            if os.path.isfile(path):
+                os.remove(path)
+            elif f not in to_keep:
+                rmtree(path)
```

Comment:
the problem with this is it will break when introducing `language_version` for `dotnet` as that allows multiple builds in a single clone

is there a way we can leverage `git clean` to do this for us?

Reply:
Ah, good point -- I had hoped we could blitz the entire git directory after the tool install.

I'll look for a `git clean`er solution

Reply:
there's also the potential that a `script` hook is defined in the same repo as well